src/main/scala/fs2/interop/cats/cats.scala

```diff
@@ -24,6 +27,15 @@ package object cats extends Instances {
     def runFoldMap[B](f: A => B)(implicit M: Monoid[B]): Free[F, B] =
       self.runFold(M.empty)((b, a) => M.combine(b, f(a)))
 
+    def runGroupByFoldMap[K, B: Monoid](f: A => K)(g: A => B): Free[F, Map[K, B]] =
+      runFoldMap(a => Map(f(a) -> g(a)))
+
+    def runGroupByFoldMonoid[K](f: A => K)(implicit M: Monoid[A]): Free[F, Map[K, A]] =
+      runFoldMap(a => Map(f(a) -> a))
+
+    def runGroupBy[X[_], K](f: A => K)(implicit A: Applicative[X], M: Monoid[X[A]]): Free[F, Map[K, X[A]]] =
```

Comment:
Hm, I'm also a bit uncomfortable requiring an `Applicative` constraint on the type constructor considering we only use it for `pure`. 

I think we should change `runGroupBy` to return `Free[F, Map[K, Vector[A]]]` -- taking the lead from `runLog` in FS2. Then, to support arbitrary collections, we can recommend use of `runGroupByFoldMap` like this:

```
Stream.range(0, 10).covary[Task].runGroupByFoldMap(_ % 2)(List(_)).run.unsafeRun
```

This is roughly the same conciseness as the completely general version of `runGroupBy` and we end up with no inference issues.